Script_Sound_Features.ipynb

```diff
@@ -0,0 +1,326 @@
+      "    Uninstalling numpy-1.26.4:\n",
+      "      Successfully uninstalled numpy-1.26.4\n",
+      "\u001b[33m  WARNING: The scripts f2py, f2py3 and f2py3.10 are installed in '/home/jupyter-acv/.local/bin' which is not on PATH.\n",
+      "  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.\u001b[0m\u001b[33m\n",
+      "\u001b[0m  Attempting uninstall: scipy\n",
+      "    Found existing installation: scipy 1.13.0\n",
+      "    Uninstalling scipy-1.13.0:\n",
+      "      Successfully uninstalled scipy-1.13.0\n",
+      "Successfully installed numpy-1.25.2 scipy-1.9.3\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "pip install scipy==1.9.3\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "9769ccc8-9e26-4bc7-8e1c-badb4b649dbc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import librosa\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from tqdm import tqdm\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "import scipy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "d90982a0-be5e-4230-957f-12559720f764",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "folder = '../../mnt/disks/songsnap/audio'\n",
+    "all_files = os.listdir(folder)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "47d3122f-b675-48a6-a319-7b3e820d3984",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generateSoundFeatures(file_name, folder):\n",
+    "    if file_name.endswith('.wav'):\n",
+    "        audio_path = os.path.join(folder, file_name)\n",
+    "\n",
+    "        base_name = os.path.splitext(file_name)[0]\n",
+    "\n",
+    "        y, sr = librosa.load(audio_path, duration=30.0)\n",
+    "    \n",
+    "        # Mel-frequency Cepstral coefficients\n",
+    "        mfccs = np.mean(librosa.feature.mfcc(y=y, sr=sr, n_mfcc=13), axis=1)\n",
+    "        # Chroma features\n",
+    "        chroma = np.mean(librosa.feature.chroma_stft(y=y, sr=sr), axis=1)\n",
+    "        # Spectral centroid\n",
+    "        spectral_centroid = np.mean(librosa.feature.spectral_centroid(y=y, sr=sr))\n",
+    "        # Spectral bandwith\n",
+    "        spectral_bandwidth = np.mean(librosa.feature.spectral_bandwidth(y=y, sr=sr))\n",
+    "        # Spectral contrast\n",
+    "        spectral_contrast = np.mean(librosa.feature.spectral_contrast(y=y, sr=sr))\n",
+    "        # Spectral rolloff\n",
+    "        spectral_rolloff = np.mean(librosa.feature.spectral_rolloff(y=y, sr=sr))\n",
+    "        # Tempo\n",
+    "        tempo, _ = librosa.beat.beat_track(y=y, sr=sr)\n",
+    "        # Zero crossing rate\n",
+    "        zero_crossing_rate = np.mean(librosa.feature.zero_crossing_rate(y))\n",
+    "    \n",
```

Comment:
I'm really curious about what these features mean lol

Reply:
together, all of these features from the librosa package provide a comprehensive summary of the acoustic characteristics of audio signals. they are commonly used for tasks like genre classification, speech recognition, etc.

Reply:
did some research and found some interesting analysis-- the chroma is related to harmonies, pitch, etc.
the relevant features of the spectrogram are captured here. the musical intuition is as follows: there are narrow-band and broad-band signals in audio processing. narrow-band signals cover only a small frequency range, while broad-band signals cover a wider range. a single flute playing a note is probably narrow, whereas some densely layered EDM might be broad. the centroid, bandwidth, contrast, etc. are features regarding the range, mean, etc. of the spectrogram